src/Extensions/Hosting.Services.Web/ApplicationBuilderExtensions.cs

```diff
@@ -23,23 +24,47 @@ public static class ApplicationBuilderExtensions
 		/// Adds the default exception handling logic that will display a developer exception page in develop and a short message during deployment
 		/// </summary>
 		/// <param name="builder">Application builder</param>
+		/// <param name="environment">Enviroment information</param>
 		/// <param name="enableCorrelationHeaderBackwardCompatibility">Set it to true if the service needs to accept a request with old style correlation</param>
-		public static IApplicationBuilder UseOmexMiddlewares(this IApplicationBuilder builder, bool enableCorrelationHeaderBackwardCompatibility)
+		public static IApplicationBuilder UseOmexMiddlewares(
+			this IApplicationBuilder builder,
+			IHostEnvironment environment,
+			bool enableCorrelationHeaderBackwardCompatibility)
 		{
 			builder
-				.UseMiddleware<ActivityEnrichmentMiddleware>()
-				.UseMiddleware<ResponseHeadersMiddleware>();
+				.UseOmexExceptionHandler(environment)
+				.UseActivityEnrichmentMiddleware()
+				.UseResponseHeadersMiddleware();
 
 			if (enableCorrelationHeaderBackwardCompatibility)
 			{
 #pragma warning disable CS0618 // We need to add this middleware to accept old correlation header
-				builder.UseMiddleware<ObsoleteCorrelationHeadersMiddleware>();
+				builder.UseObsoleteCorrelationHeadersMiddleware();
 #pragma warning restore CS0618
 			}
 
 			return builder;
 		}
 
+		/// <summary>
+		/// Add middleware to enrich request activity with Result, SubType and Metadata
+		/// </summary>
+		public static IApplicationBuilder UseActivityEnrichmentMiddleware(this IApplicationBuilder builder) =>
+			builder.UseMiddleware<ActivityEnrichmentMiddleware>();
+
+		/// <summary>
+		/// Add middleware that adds Omex headers to responses, like MachineId and BuildVersion
+		/// </summary>
+		public static IApplicationBuilder UseResponseHeadersMiddleware(this IApplicationBuilder builder) =>
+			builder.UseMiddleware<ResponseHeadersMiddleware>();
+
+		/// <summary>
+		/// Add middleware that adds Omex headers to responses, like MachineId and BuildVersion
+		/// </summary>
+		[Obsolete("Use it only if you need to comunicate with services that use old correlation", false)]
+		public static IApplicationBuilder UseObsoleteCorrelationHeadersMiddleware(this IApplicationBuilder builder) =>
```

Comment:
Do these need to be public?

Reply:
Unfortunately yes, it's for the case when you need want to add only this middleware